engine/src/cubos/engine/settings/plugin.cpp

```diff
@@ -0,0 +1,85 @@
+#include <cubos/core/data/fs/file_system.hpp>
+#include <cubos/core/data/fs/standard_archive.hpp>
+#include <cubos/core/data/json_deserializer.hpp>
+
+#include <cubos/engine/settings/plugin.hpp>
+
+using cubos::core::data::File;
+using cubos::core::data::FileSystem;
+using cubos::core::data::JSONDeserializer;
+using cubos::core::data::StandardArchive;
+using cubos::core::ecs::Read;
+using cubos::core::ecs::Write;
+
+using namespace cubos::engine;
+
+static Settings loadFromArguments(const Arguments& args)
+{
+    Settings settings{};
+
+    for (const auto& arg : args.value)
+    {
+        auto token = arg.find('=');
+        if (token == std::string::npos)
+        {
+            CUBOS_ERROR("Could not parse argument: missing '=' token");
+            break;
+        }
+
+        auto settingName = arg.substr(0, token);
+        auto settingValue = arg.substr(token + 1);
+        if (settingValue.empty())
+        {
+            CUBOS_ERROR("Setting '{}' must have value", settingName);
+            break;
+        }
+
+        settings.setString(settingName, settingValue);
+    }
+
+    return settings;
+}
+
+static Settings loadFromFile(std::string_view path)
+{
+    // Mount the real settings file as `/settings.json`.
+    FileSystem::mount("/settings.json",
+                      std::make_unique<StandardArchive>(path, false /*isDirectory*/, false /*readOnly*/));
```

Comment:
does this behave well if there is no settings file at path?

Reply:
Yeah, `StandardArchive` docs states that it creates an empty file if there isn't one.